jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -0,0 +1,204 @@
+/* Copyright (C) 2015 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Displays post-deployment information for usage and quickstart.
+  Shown when a charm or bundle is added to the canvas, and on deploy.
+*/
+class PostDeployment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.variables = {
+      details_link: {
+        onClick: this._handleViewDetails.bind(this),
+        content: '<span role="button" ' +
+          'class="link" ' +
+          'data-variable="details_link">View details</span>'
+      },
+      requires_cli_link: {
+        content: '<a ' +
+          'href="https://jujucharms.com/docs/stable/reference-install" ' +
+          'target="_blank">Juju CLI client</a>'
+      }
+    };
+    this.state= {
+      content: null,
+      metadata: {}
+    };
+  }
+
+  componentDidMount() {
+    const files = this.props.files;
+
+    // We only want to try and get the usage.md file if there's one there.
+    // Unfortunately, the only place we can easily know whether the file
+    // exists or not is from the entity details page.
+    // The + icon in search results will make an api request for the file
+    // and 404 if it's not there.
+    if (!files || (files && files.indexOf('usage.md') !== -1)) {
+      this.props.getFile(
+        this.props.entityId,
+        'usage.md',
+        this._getUsageCallback.bind(this)
+      );
+    }
+  }
+
+  /**
+    Callback when usage.md is retrieved.
+
+    @param {Object} error Error if there is one.
+    @param {String} usageContents The contents of the file.
+  */
+  _getUsageCallback(error, usageContents) {
+    if (usageContents && usageContents.substring(0, 1) !== '{') {
```

Comment:
Please handle the error here, for instance, by sending a notification to the user.